lib/dependency-manager-adapters/npm.js

```diff
@@ -7,13 +7,37 @@ var path = require('path');
 var extend = require('extend');
 var debug = require('debug')('ember-try:dependency-manager-adapter:npm');
 var rimraf = RSVP.denodeify(require('rimraf'));
+var execFileSync = require('child_process').execFileSync;
 
 module.exports = CoreObject.extend({
   init: function() {
     this._super.apply(this, arguments);
     this.run = this.run || require('../utils/run');
+    this.adapterInit();
   },
-  configKey: 'npm',
+  adapterInit() {
+    this.yarnLockPresent = fs.existsSync(path.join(this.cwd, this.yarnLock));
+    let yarnManagerPresent;
+    try {
+      execFileSync('yarn', [].concat(['--version']));
+      yarnManagerPresent = true;
+    } catch (e) {
+      yarnManagerPresent = false;
+    }
+    if (this.yarnLockPresent || yarnManagerPresent) {
+      this.useYarn = true;
+      this.configKey = 'yarn';
```

Comment:
The `configKey` property is used here to determine the config key used in the dependency set object https://github.com/ember-cli/ember-try/blob/master/lib/dependency-manager-adapters/npm.js#L26, which eventually originates in the `config/ember-try.js`

Some of my thoughts on support of yarn in this way (rather than through a separate adapter):
- Users with `yarn` installed get to use and benefit from `yarn install` without changing ember-try config. 
- Users with a `yarn.lock` and/or `yarn` installed (TBD elsewhere in these comments) have yarn used for their ember-try scenarios. The config stays under `npm` because even though the end user is using `yarn`, `package.json` and node_modules are the source of the dependencies, since we will ignore the `yarn.lock` (We _want_/_need_ dependencies to float for the purpose of ember-try).
- No duplication of generated configs just to support `yarn` as well, when the contents of the config will be the same. (Such as that from `ember-cli` blueprint or `ember-try-config`

If wanting to display `yarn` in the table rather than `npm` (I'm not 100% positive this is necessary, users may expect the table to match their config), at that point we could check `this.useYarn`.